diff --git a/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift b/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
index b1c9f8bda4db3053d8449b8b6475166368ffbea0..6464afcaa392b3ad0ce17ef07cdfefa8399985cd 100644
--- a/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
+++ b/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
@@ -265,7 +265,10 @@ extension UIHostingController {
       case .all:
         self.safeAreaRegions.remove(.all)
       case .keyboard:
+#if os(iOS)
         self.safeAreaRegions.remove(.keyboard)
+#endif
+        break
       }
     } else {
       // For older versions
@@ -290,11 +293,13 @@ extension UIHostingController {
                               imp_implementationWithBlock(safeAreaInsets), method_getTypeEncoding(method))
           }

+#if os(iOS)
           if let method = class_getInstanceMethod(viewClass, NSSelectorFromString("keyboardWillShowWithNotification:")) {
               let keyboardWillShow: @convention(block) (AnyObject, AnyObject) -> Void = { _, _ in }
               class_addMethod(viewSubclass, NSSelectorFromString("keyboardWillShowWithNotification:"),
                               imp_implementationWithBlock(keyboardWillShow), method_getTypeEncoding(method))
           }
+#endif
           objc_registerClassPair(viewSubclass)
           object_setClass(view, viewSubclass)
         }
